### PR TITLE
Add crc header in put blob response

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azblob/blockblob/client_test.go [hunk 1]
```diff
@@ -357,6 +357,36 @@ func (s *BlockBlobUnrecordedTestsSuite) TestBlockBlobClientConnectionString() {
...
+	_require.NoError(err)
+
+	contentSize := 4 * 1024 // 4 KB
+	r, _ := testcommon.GetDataAndReader(testName, contentSize)
+	rsc := streaming.NopCloser(r)
+
+	resp, err := bbClient.Upload(context.Background(), rsc, nil)
+	_require.NoError(err)
+	_require.NotNil(resp)
+	_require.NotNil(resp.ContentCRC64)
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/blockblob/client_test.go:387]: Can you also add a validation step where the crc64 value returned in the response is compared against the one computed for the source data.

**Review by souravgupta-msft — APPROVED** [review]

File: sdk/storage/azblob/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 ### Breaking Changes
 
 ### Bugs Fixed
+* Fixed a bug to ensure crc64 response header is returned in Put Blob request. Fixes [#23372](https://github.com/Azure/azure-sdk-for-go/issues/23372).
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/CHANGELOG.md:10]: Technically this should be listed as a feature (new public surface area is always a feature).

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/23372

File: sdk/storage/azblob/blockblob/client_test.go (hunk 1 above)
Comment (tanyasethi-msft) [reply sdk/storage/azblob/blockblob/client_test.go:390]: Added the check

File: sdk/storage/azblob/CHANGELOG.md (hunk 2 above)
Comment (tanyasethi-msft) [reply sdk/storage/azblob/CHANGELOG.md]: Moved to Features list

**Pull request merged**